### PR TITLE
boost elevation UI for free cursor tool

### PR DESCRIPTION
**Pull request by kianzarrin**
I have added quite a few features. I hope I did not overdo it!
### Functional changes:
- [Home] resets elevation to zero. if `Absolute Elevation` is checked it sets elevation to terrain height. :)
- elevation step  ([Page-Up] / [Page-Down]) is read from FineRoadTool or NetTool (if FRT is disabled/absent)
  - using the elevation step button/slider does not unfocus/disable Roundabout builder which is nice and handy (I did not have to write any code to ensure this).  see screenshot
 - whenever Free cursor window become visible (close and open window | press free cursor tool) it reads elevation from FRT/NetTool. It is correct even if `Absolute Elevation` is checked.
  - To test this Select Road tool and raise elevation to build a elevated road. then select Free cursor tool and see what is the initial elevation (its same as road tool)
- checking `Absolute Elevation` samples terrain height at center of screen and increase elevation field such that the roundabout stays at the same vertical position (`new value = old value + centerTerrainY`). 
  - unchecking `Absolute Elevation` sets elevation to whatever value it was before checking `Absolute Elevation`

Screenshot (771)

### other code changes
- changed `NumericField.Value` from `float` to `int`. this is because it is impossible to set float value and trying to do so results in silent failure.
- commented out this code. Its redundant right?
```C#
// absoluteElevation.isChecked = EllipseTool.Instance.ControlVertices; // TODO this is redundant right?
absoluteElevation.relativePosition = new Vector3(8, cumulativeHeight);
absoluteElevation.isChecked = false;
```

# Playing around:
I added quite a few features. I recommend to play with these new features so that you know how your tool works. Test these with and without FRT :

### PageUp/PageDown/Home tests:
- [ ] change elevation step in NetTool/FRT. then press page up/page down in Free cursor tool to verify changing elevation step is working
- [ ] UNCHECK Absolute Elevation and raise elevation. Then press home button to verify elevation becomes 0.
- [ ] CHECK Absolute Elevation and raise elevation. Then press home button to verify relative elevation is at ground level.

### Auto Initialize tests
- [ ] UNCHECK Absolute Elevation in Free Cursor Tool. Set elevation to some value. CHECK Absolute Elevation  and verify that round about is at relatively the same height at center of screen.
- [ ] UNCHECK Absolute Elevation again and verify its previous value is resotred
- [ ] Close Roundabout Builder panel. In Road Tool raise road elevation. Then open Free Cursor Tool and validate initial elevation is same as Road Tool.
- CHECK Absolute Elevation in Free Cursor Tool then close the panel. Go to Road Tool and raise road elevation. Open Free Cursor Tool and verify that Elevation field is initialized such that roundabout is elevated above ground at same height at screen center.

### CODE REVIEW

**Comment by kianzarrin** [comment]
I am fixing the conflicts.
EDIT: I thought I have updated to latest version but I made a mistake.

**Comment by kianzarrin** [comment]
I am still working on this. Trying to auto detect startup elevation.

**Comment by kianzarrin** [comment]
DONE! I am ready for review.

**Comment by Strdate** [comment]
Many thanks! This time it probably won't be as quick though :)

**Comment by Strdate** [comment]
Many thanks! This time it probably won't be as quick though :)
First I would like to finish the exception report mod I was showing on discord as this might resolve some problems users are experiencing with roundabout builder. With that I would like to go back to this.

File: RoundaboutBuilder/UI/Windows.cs [hunk 1]
```diff
@@ -153,17 +164,42 @@ public override void Start()
             absoluteElevation.name = "RAB_absoluteElevation";
             absoluteElevation.label.text = "Absolute elevation";
             absoluteElevation.tooltip = "Elevation will be measured from zero level instead of terrain level";
-            absoluteElevation.isChecked = EllipseTool.Instance.ControlVertices;
+            // absoluteElevation.isChecked = EllipseTool.Instance.ControlVertices; // TODO is this necessary
```
Comment (Strdate) [inline RoundaboutBuilder/UI/Windows.cs]: Yeah, this should be redundant

**Comment by Strdate** [comment]
I looked through it and it seems fine. It is now released.
Thanks :)

**Pull request merged**